### ISSUE TITLE
Supporting custom domains

### ISSUE THREAD

**Issue by hjertnes** [jointwt/twtxt#37]
Have you considered adding support for pointing to custom domains?

**Comment by prologic** [comment]
What do you mean by this? There are plans in the works eventually to have  "paid hosted instnaces" at some point for the rest of us (_those who either can't, don't want to have to manage one, or have no idea how_). Is this what you mean?

**Comment by hjertnes** [comment]
Nope. 

Most hosted blogging systems (or even github pages) support that you point a domain where you manage DNS to them. 

For example, I would never want to move to any twtxt thing where it would break my current URL (https://hjertnes.social/twtxt.txt)

**Comment by prologic** [comment]
Ahh yes I see what you mean now...

Can you have a think about how this work work exactly?

Your example (https://hjertnes.social/twtxt.txt) is a `text/plain` web resource already, so its unclear to me how a user account's custom domain would work here?

**Comment by hjertnes** [comment]
Yeah, but if I were to start using twtxt.net, I would like to point my domain against twtxt.net without that url breaking

**Comment by prologic** [comment]
> Yeah, but if I were to start using twtxt.net, I would like to point my domain against twtxt.net without that url breaking

No I totally get that, in fact its a good idea! Let's figure out the details of "how" :)

**Comment by lwojcik** [comment]
As a typical user, a pattern I've observed with different service providers is adding my domain in the settings tab and adding a relevant CNAME record in my DNS settings pointing to a target server supposed to resolve it.

Using a subdomain lwojcik.mydomain.ltd I'd expect it to resolve any path specific to my twtxt profile. Examples:

* lwojcik.mydomain.tld/ => https://twtxt.net/user/lukem 
* lwojcik.mydomain.tld/twtxt.txt => https://twtxt.net/user/lukem/twtxt.txt
* lwojcik.mydomain.tld/followers => https://twtxt.net/user/lukem/followers

As a potential instance owner I feel this will probably require generating SSL certs to sign custom domains and / or wildcard subdomains, but I'm not overly familiar with this area.

**Comment by prologic** [comment]
@lwojcik Ahh brilliant. This it the kind of detail we need. Technically this would not be hard to do, you'd effectively need a "special router" that handles "vhosts" effectively based on user settings.

Re as a user setting up wildcard certs is pretty easy to do with Let's  Encrypt and Traefik also makes this trivial to setup and get right.

**Comment by prologic** [comment]
Now with the launch of [Twt.sockal](https://twt.social) (#87) I've been giving this a lot of though. I designed the backend for `twt.social` to support this; but there are some complications.

- How/Where do you terminate TLS?

Its easy enough to point your domain via a CNAME somewhere; but now I have to think about the ingress load balancer as well and potentially reconfiguring it to support another domain for TLS (_e.g: Let's Encrypt_). Thoughts? Maybe there's an easier way?

**Comment by prologic** [comment]
Still planned

**Comment by prologic** [comment]
Bah this issue was closed in error by the stupid bot 😂
